tensorboard dependency grpcio 1.40.0 is not supported for python 2.7

```
Run python -m pip install --upgrade -r build/test-requirements.txt
DEPRECATION: Python 2.7 reached the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 is no longer maintained. pip 21.0 will drop support for Python 2.7 in January 2021. More details about Python 2 support in pip can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support pip 21.0 will remove support for this functionality.
Ignoring bandit: markers 'python_version >= "3.5"' don't match your environment
Ignoring black: markers 'python_version > "2.7"' don't match your environment
Ignoring prospector: markers 'python_version > "2.7"' don't match your environment
Ignoring pytest: markers 'python_version > "2.7"' don't match your environment
Ignoring fastapi: markers 'python_version > "2.7"' don't match your environment
Ignoring uvicorn: markers 'python_version > "2.7"' don't match your environment
Collecting flake8
  Downloading flake8-3.9.2-py2.py3-none-any.whl (73 kB)
Collecting autopep8
  Downloading autopep8-1.5.7-py2.py3-none-any.whl (45 kB)
Collecting yapf
  Downloading yapf-0.31.0-py2.py3-none-any.whl (185 kB)
Collecting pylint
  Downloading pylint-1.9.5-py2.py3-none-any.whl (696 kB)
Collecting pycodestyle
  Downloading pycodestyle-2.7.0-py2.py3-none-any.whl (41 kB)
Collecting pydocstyle
  Downloading pydocstyle-3.0.0-py2-none-any.whl (32 kB)
Collecting rope
  Downloading rope-0.19.0.tar.gz (252 kB)
Collecting flask
  Downloading Flask-1.1.4-py2.py3-none-any.whl (94 kB)
Collecting django
  Downloading Django-1.11.29-py2.py3-none-any.whl (6.9 MB)
Collecting isort
  Downloading isort-4.3.21-py2.py3-none-any.whl (42 kB)
Collecting tensorboard
  Downloading tensorboard-2.1.0-py2-none-any.whl (3.8 MB)
Collecting pytest==4.6.9
  Downloading pytest-4.6.9-py2.py3-none-any.whl (231 kB)
Collecting py==1.10.0
  Downloading py-1.10.0-py2.py3-none-any.whl (97 kB)
Collecting pathlib2>=2.2.0
  Downloading pathlib2-2.3.6-py2.py3-none-any.whl (17 kB)
Collecting prospector==1.2.0
  Downloading prospector-1.2.0.tar.gz (55 kB)
Collecting configparser; python_version < "3.2"
  Downloading configparser-4.0.2-py2.py3-none-any.whl (22 kB)
Collecting typing; python_version < "3.5"
  Downloading typing-3.10.0.0-py2-none-any.whl (26 kB)
Collecting functools32; python_version < "3.2"
  Downloading functools32-3.2.3-2.tar.gz (31 kB)
Collecting mccabe<0.7.0,>=0.6.0
  Downloading mccabe-0.6.1-py2.py3-none-any.whl (8.6 kB)
Collecting importlib-metadata; python_version < "3.8"
  Downloading importlib_metadata-2.1.1-py2.py3-none-any.whl (10 kB)
Collecting pyflakes<2.4.0,>=2.3.0
  Downloading pyflakes-2.3.1-py2.py3-none-any.whl (68 kB)
Collecting enum34; python_version < "3.4"
  Downloading enum34-1.1.10-py2-none-any.whl (11 kB)
Collecting toml
  Downloading toml-0.10.2-py2.py3-none-any.whl (16 kB)
Collecting backports.functools-lru-cache; python_version == "2.7"
  Downloading backports.functools_lru_cache-1.6.4-py2.py3-none-any.whl (5.9 kB)
Collecting colorama; sys_platform == "win32"
  Downloading colorama-0.4.4-py2.py3-none-any.whl (16 kB)
Collecting astroid<2.0,>=1.6
  Downloading astroid-1.6.6-py2.py3-none-any.whl (305 kB)
Collecting singledispatch; python_version < "3.4"
  Downloading singledispatch-3.7.0-py2.py3-none-any.whl (9.2 kB)
Collecting six
  Downloading six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting snowballstemmer
  Downloading snowballstemmer-2.1.0-py2.py3-none-any.whl (93 kB)
Collecting itsdangerous<2.0,>=0.24
  Downloading itsdangerous-1.1.0-py2.py3-none-any.whl (16 kB)
Collecting click<8.0,>=5.1
  Downloading click-7.1.2-py2.py3-none-any.whl (82 kB)
Collecting Jinja2<3.0,>=2.10.1
  Downloading Jinja2-2.11.3-py2.py3-none-any.whl (125 kB)
Collecting Werkzeug<2.0,>=0.15
  Downloading Werkzeug-1.0.1-py2.py3-none-any.whl (298 kB)
Collecting pytz
  Downloading pytz-2021.1-py2.py3-none-any.whl (510 kB)
Collecting futures; python_version < "3.2"
  Downloading futures-3.3.0-py2-none-any.whl (16 kB)
Collecting numpy>=1.12.0
  Downloading numpy-1.16.6-cp27-cp27m-win_amd64.whl (11.9 MB)
Collecting requests<3,>=2.21.0
  Downloading requests-2.26.0-py2.py3-none-any.whl (62 kB)
Collecting grpcio>=1.24.3
  Downloading grpcio-1.40.0.tar.gz (21.3 MB)
Collecting absl-py>=0.4
  Downloading absl-py-0.13.0.tar.gz (114 kB)
Collecting wheel; python_version < "3"
  Downloading wheel-0.37.0-py2.py3-none-any.whl (35 kB)
Collecting protobuf>=3.6.0
  Downloading protobuf-3.17.3-py2.py3-none-any.whl (173 kB)
Collecting markdown>=2.6.8
  Downloading Markdown-3.1.1-py2.py3-none-any.whl (87 kB)
Collecting google-auth<2,>=1.6.3
  Downloading google_auth-1.35.0-py2.py3-none-any.whl (152 kB)
Collecting google-auth-oauthlib<0.5,>=0.4.1
  Downloading google_auth_oauthlib-0.4.1-py2.py3-none-any.whl (18 kB)
Requirement already satisfied, skipping upgrade: setuptools>=41.0.0 in c:\hostedtoolcache\windows\python\2.7.18\x64\lib\site-packages (from tensorboard->-r build/test-requirements.txt (line 21)) (41.2.0)
Collecting attrs>=17.4.0
  Downloading attrs-21.2.0-py2.py3-none-any.whl (53 kB)
Collecting more-itertools<6.0.0,>=4.0.0; python_version <= "2.7"
  Downloading more_itertools-5.0.0-py2-none-any.whl (52 kB)
Collecting pluggy<1.0,>=0.12
  Downloading pluggy-0.13.1-py2.py3-none-any.whl (18 kB)
Collecting packaging
  Downloading packaging-20.9-py2.py3-none-any.whl (40 kB)
Collecting wcwidth
  Downloading wcwidth-0.2.5-py2.py3-none-any.whl (30 kB)
Collecting funcsigs>=1.0; python_version < "3.0"
  Downloading funcsigs-1.0.2-py2.py3-none-any.whl (17 kB)
Collecting atomicwrites>=1.0
  Downloading atomicwrites-1.4.0-py2.py3-none-any.whl (6.8 kB)
Collecting scandir; python_version < "3.5"
  Downloading scandir-1.10.0-cp27-cp27m-win_amd64.whl (20 kB)
Collecting pylint-plugin-utils>=0.2.6
  Downloading pylint-plugin-utils-0.6.tar.gz (10 kB)
Collecting pylint-celery==0.3
  Downloading pylint-celery-0.3.tar.gz (1.9 kB)
Collecting pylint-flask==0.6
  Downloading pylint-flask-0.6.tar.gz (10.0 kB)
Collecting requirements-detector>=0.6
  Downloading requirements-detector-0.7.tar.gz (8.6 kB)
Collecting setoptconf>=0.2.0
  Downloading setoptconf-0.2.0.tar.gz (13 kB)
Collecting dodgy>=0.1.9
  Downloading dodgy-0.2.1.tar.gz (4.9 kB)
Collecting pyyaml
  Downloading PyYAML-5.4.1-cp27-cp27m-win_amd64.whl (204 kB)
Collecting pep8-naming<=0.4.1,>=0.3.3
  Downloading pep8_naming-0.4.1-py2.py3-none-any.whl (8.1 kB)
Collecting pylint-django<0.9
  Downloading pylint-django-0.8.1.tar.gz (26 kB)
Collecting contextlib2; python_version < "3"
  Downloading contextlib2-0.6.0.post1-py2.py3-none-any.whl (9.8 kB)
Collecting zipp>=0.5
  Downloading zipp-1.2.0-py2.py3-none-any.whl (4.8 kB)
Collecting wrapt
  Downloading wrapt-1.12.1.tar.gz (27 kB)
Collecting lazy-object-proxy
  Downloading lazy_object_proxy-1.6.0-cp27-cp27m-win_amd64.whl (20 kB)
Collecting MarkupSafe>=0.23
  Downloading MarkupSafe-1.1.1-cp27-cp27m-win_amd64.whl (13 kB)
Collecting chardet<5,>=3.0.2; python_version < "3"
  Downloading chardet-4.0.0-py2.py3-none-any.whl (178 kB)
Collecting idna<3,>=2.5; python_version < "3"
  Downloading idna-2.10-py2.py3-none-any.whl (58 kB)
Collecting urllib3<1.27,>=1.21.1
  Downloading urllib3-1.26.6-py2.py3-none-any.whl (138 kB)
Collecting certifi>=2017.4.17
  Downloading certifi-2021.5.30-py2.py3-none-any.whl (145 kB)
Collecting pyasn1-modules>=0.2.1
  Downloading pyasn1_modules-0.2.8-py2.py3-none-any.whl (155 kB)
Collecting rsa<4.6; python_version < "3.6"
  Downloading rsa-4.5-py2.py3-none-any.whl (36 kB)
Collecting cachetools<5.0,>=2.0.0
  Downloading cachetools-3.1.1-py2.py3-none-any.whl (11 kB)
Collecting requests-oauthlib>=0.7.0
  Downloading requests_oauthlib-1.3.0-py2.py3-none-any.whl (23 kB)
Collecting pyparsing>=2.0.2
  Downloading pyparsing-2.4.7-py2.py3-none-any.whl (67 kB)
Collecting pyasn1<0.5.0,>=0.4.6
  Downloading pyasn1-0.4.8-py2.py3-none-any.whl (77 kB)
Collecting oauthlib>=3.0.0
  Downloading oauthlib-3.1.0-py2.py3-none-any.whl (147 kB)
Using legacy 'setup.py install' for rope, since package 'wheel' is not installed.
Using legacy 'setup.py install' for prospector, since package 'wheel' is not installed.
Using legacy 'setup.py install' for functools32, since package 'wheel' is not installed.
Using legacy 'setup.py install' for grpcio, since package 'wheel' is not installed.
Using legacy 'setup.py install' for absl-py, since package 'wheel' is not installed.
Using legacy 'setup.py install' for pylint-plugin-utils, since package 'wheel' is not installed.
Using legacy 'setup.py install' for pylint-celery, since package 'wheel' is not installed.
Using legacy 'setup.py install' for pylint-flask, since package 'wheel' is not installed.
Using legacy 'setup.py install' for requirements-detector, since package 'wheel' is not installed.
Using legacy 'setup.py install' for setoptconf, since package 'wheel' is not installed.
Using legacy 'setup.py install' for dodgy, since package 'wheel' is not installed.
Using legacy 'setup.py install' for pylint-django, since package 'wheel' is not installed.
Using legacy 'setup.py install' for wrapt, since package 'wheel' is not installed.
Installing collected packages: pycodestyle, configparser, typing, functools32, mccabe, contextlib2, zipp, six, scandir, pathlib2, importlib-metadata, pyflakes, enum34, flake8, toml, autopep8, yapf, backports.functools-lru-cache, colorama, wrapt, lazy-object-proxy, singledispatch, astroid, futures, isort, pylint, snowballstemmer, pydocstyle, rope, itsdangerous, click, MarkupSafe, Jinja2, Werkzeug, flask, pytz, django, numpy, chardet, idna, urllib3, certifi, requests, grpcio, absl-py, wheel, protobuf, markdown, pyasn1, pyasn1-modules, rsa, cachetools, google-auth, oauthlib, requests-oauthlib, google-auth-oauthlib, tensorboard, attrs, more-itertools, pluggy, pyparsing, packaging, wcwidth, funcsigs, atomicwrites, py, pytest, pylint-plugin-utils, pylint-celery, pylint-flask, requirements-detector, setoptconf, dodgy, pyyaml, pep8-naming, pylint-django, prospector
    Running setup.py install for functools32: started
    Running setup.py install for functools32: finished with status 'done'
    Running setup.py install for wrapt: started
    Running setup.py install for wrapt: finished with status 'done'
    Running setup.py install for rope: started
    Running setup.py install for rope: finished with status 'done'
    Running setup.py install for grpcio: started
    Running setup.py install for grpcio: finished with status 'error'
    ERROR: Command errored out with exit status 1:
     command: 'C:\hostedtoolcache\windows\Python\2.7.18\x64\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'c:\\users\\runner~1\\appdata\\local\\temp\\pip-install-yixmuu\\grpcio\\setup.py'"'"'; __file__='"'"'c:\\users\\runner~1\\appdata\\local\\temp\\pip-install-yixmuu\\grpcio\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'c:\users\runner~1\appdata\local\temp\pip-record-ezdbm8\install-record.txt' --single-version-externally-managed --compile --install-headers 'C:\hostedtoolcache\windows\Python\2.7.18\x64\Include\grpcio'
         cwd: c:\users\runner~1\appdata\local\temp\pip-install-yixmuu\grpcio\
    Complete output (150 lines):
    ('ASM Builds for BoringSSL currently not supported on:', 'win-amd64')
    Found cython-generated files...
    running install
    running build
    running build_py
    running build_project_metadata
    creating python_build
    creating python_build\lib.win-amd64-2.7
    creating python_build\lib.win-amd64-2.7\grpc
    copying src\python\grpcio\grpc\_auth.py -> python_build\lib.win-amd64-2.7\grpc
    copying src\python\grpcio\grpc\_channel.py -> python_build\lib.win-amd64-2.7\grpc
    copying src\python\grpcio\grpc\_common.py -> python_build\lib.win-amd64-2.7\grpc
    copying src\python\grpcio\grpc\_compression.py -> python_build\lib.win-amd64-2.7\grpc
    copying src\python\grpcio\grpc\_grpcio_metadata.py -> python_build\lib.win-amd64-2.7\grpc
    copying src\python\grpcio\grpc\_interceptor.py -> python_build\lib.win-amd64-2.7\grpc
    copying src\python\grpcio\grpc\_plugin_wrapping.py -> python_build\lib.win-amd64-2.7\grpc
    copying src\python\grpcio\grpc\_runtime_protos.py -> python_build\lib.win-amd64-2.7\grpc
    copying src\python\grpcio\grpc\_server.py -> python_build\lib.win-amd64-2.7\grpc
    copying src\python\grpcio\grpc\_simple_stubs.py -> python_build\lib.win-amd64-2.7\grpc
    copying src\python\grpcio\grpc\_utilities.py -> python_build\lib.win-amd64-2.7\grpc
    copying src\python\grpcio\grpc\__init__.py -> python_build\lib.win-amd64-2.7\grpc
    creating python_build\lib.win-amd64-2.7\grpc\aio
    copying src\python\grpcio\grpc\aio\_base_call.py -> python_build\lib.win-amd64-2.7\grpc\aio
    copying src\python\grpcio\grpc\aio\_base_channel.py -> python_build\lib.win-amd64-2.7\grpc\aio
    copying src\python\grpcio\grpc\aio\_base_server.py -> python_build\lib.win-amd64-2.7\grpc\aio
    copying src\python\grpcio\grpc\aio\_call.py -> python_build\lib.win-amd64-2.7\grpc\aio
    copying src\python\grpcio\grpc\aio\_channel.py -> python_build\lib.win-amd64-2.7\grpc\aio
    copying src\python\grpcio\grpc\aio\_interceptor.py -> python_build\lib.win-amd64-2.7\grpc\aio
    copying src\python\grpcio\grpc\aio\_metadata.py -> python_build\lib.win-amd64-2.7\grpc\aio
    copying src\python\grpcio\grpc\aio\_server.py -> python_build\lib.win-amd64-2.7\grpc\aio
    copying src\python\grpcio\grpc\aio\_typing.py -> python_build\lib.win-amd64-2.7\grpc\aio
    copying src\python\grpcio\grpc\aio\_utils.py -> python_build\lib.win-amd64-2.7\grpc\aio
    copying src\python\grpcio\grpc\aio\__init__.py -> python_build\lib.win-amd64-2.7\grpc\aio
    creating python_build\lib.win-amd64-2.7\grpc\beta
    copying src\python\grpcio\grpc\beta\implementations.py -> python_build\lib.win-amd64-2.7\grpc\beta
    copying src\python\grpcio\grpc\beta\interfaces.py -> python_build\lib.win-amd64-2.7\grpc\beta
    copying src\python\grpcio\grpc\beta\utilities.py -> python_build\lib.win-amd64-2.7\grpc\beta
    copying src\python\grpcio\grpc\beta\_client_adaptations.py -> python_build\lib.win-amd64-2.7\grpc\beta
    copying src\python\grpcio\grpc\beta\_metadata.py -> python_build\lib.win-amd64-2.7\grpc\beta
    copying src\python\grpcio\grpc\beta\_server_adaptations.py -> python_build\lib.win-amd64-2.7\grpc\beta
    copying src\python\grpcio\grpc\beta\__init__.py -> python_build\lib.win-amd64-2.7\grpc\beta
    creating python_build\lib.win-amd64-2.7\grpc\experimental
    copying src\python\grpcio\grpc\experimental\gevent.py -> python_build\lib.win-amd64-2.7\grpc\experimental
    copying src\python\grpcio\grpc\experimental\session_cache.py -> python_build\lib.win-amd64-2.7\grpc\experimental
    copying src\python\grpcio\grpc\experimental\__init__.py -> python_build\lib.win-amd64-2.7\grpc\experimental
    creating python_build\lib.win-amd64-2.7\grpc\framework
    copying src\python\grpcio\grpc\framework\__init__.py -> python_build\lib.win-amd64-2.7\grpc\framework
    creating python_build\lib.win-amd64-2.7\grpc\_cython
    copying src\python\grpcio\grpc\_cython\__init__.py -> python_build\lib.win-amd64-2.7\grpc\_cython
    creating python_build\lib.win-amd64-2.7\grpc\experimental\aio
    copying src\python\grpcio\grpc\experimental\aio\__init__.py -> python_build\lib.win-amd64-2.7\grpc\experimental\aio
    creating python_build\lib.win-amd64-2.7\grpc\framework\common
    copying src\python\grpcio\grpc\framework\common\cardinality.py -> python_build\lib.win-amd64-2.7\grpc\framework\common
    copying src\python\grpcio\grpc\framework\common\style.py -> python_build\lib.win-amd64-2.7\grpc\framework\common
    copying src\python\grpcio\grpc\framework\common\__init__.py -> python_build\lib.win-amd64-2.7\grpc\framework\common
    creating python_build\lib.win-amd64-2.7\grpc\framework\foundation
    copying src\python\grpcio\grpc\framework\foundation\abandonment.py -> python_build\lib.win-amd64-2.7\grpc\framework\foundation
    copying src\python\grpcio\grpc\framework\foundation\callable_util.py -> python_build\lib.win-amd64-2.7\grpc\framework\foundation
    copying src\python\grpcio\grpc\framework\foundation\future.py -> python_build\lib.win-amd64-2.7\grpc\framework\foundation
    copying src\python\grpcio\grpc\framework\foundation\logging_pool.py -> python_build\lib.win-amd64-2.7\grpc\framework\foundation
    copying src\python\grpcio\grpc\framework\foundation\stream.py -> python_build\lib.win-amd64-2.7\grpc\framework\foundation
    copying src\python\grpcio\grpc\framework\foundation\stream_util.py -> python_build\lib.win-amd64-2.7\grpc\framework\foundation
    copying src\python\grpcio\grpc\framework\foundation\__init__.py -> python_build\lib.win-amd64-2.7\grpc\framework\foundation
    creating python_build\lib.win-amd64-2.7\grpc\framework\interfaces
    copying src\python\grpcio\grpc\framework\interfaces\__init__.py -> python_build\lib.win-amd64-2.7\grpc\framework\interfaces
    creating python_build\lib.win-amd64-2.7\grpc\framework\interfaces\base
    copying src\python\grpcio\grpc\framework\interfaces\base\base.py -> python_build\lib.win-amd64-2.7\grpc\framework\interfaces\base
    copying src\python\grpcio\grpc\framework\interfaces\base\utilities.py -> python_build\lib.win-amd64-2.7\grpc\framework\interfaces\base
    copying src\python\grpcio\grpc\framework\interfaces\base\__init__.py -> python_build\lib.win-amd64-2.7\grpc\framework\interfaces\base
    creating python_build\lib.win-amd64-2.7\grpc\framework\interfaces\face
    copying src\python\grpcio\grpc\framework\interfaces\face\face.py -> python_build\lib.win-amd64-2.7\grpc\framework\interfaces\face
    copying src\python\grpcio\grpc\framework\interfaces\face\utilities.py -> python_build\lib.win-amd64-2.7\grpc\framework\interfaces\face
    copying src\python\grpcio\grpc\framework\interfaces\face\__init__.py -> python_build\lib.win-amd64-2.7\grpc\framework\interfaces\face
    creating python_build\lib.win-amd64-2.7\grpc\_cython\_cygrpc
    copying src\python\grpcio\grpc\_cython\_cygrpc\__init__.py -> python_build\lib.win-amd64-2.7\grpc\_cython\_cygrpc
    creating python_build\lib.win-amd64-2.7\grpc\_cython\_credentials
    copying src\python\grpcio\grpc\_cython\_credentials\roots.pem -> python_build\lib.win-amd64-2.7\grpc\_cython\_credentials
    running build_ext
    Non-fatal exception:Traceback (most recent call last):
      File "c:\users\runner~1\appdata\local\temp\pip-install-yixmuu\grpcio\src\python\grpcio\commands.py", line 239, in compiler_ok_with_extra_std
        stderr=subprocess.PIPE)
      File "C:\hostedtoolcache\windows\Python\2.7.18\x64\lib\subprocess.py", line 394, in __init__
        errread, errwrite)
      File "C:\hostedtoolcache\windows\Python\2.7.18\x64\lib\subprocess.py", line 644, in _execute_child
        startupinfo)
    WindowsError: [Error 2] The system cannot find the file specified
    
    Found cython-generated files...
    building 'grpc._cython.cygrpc' extension
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "c:\users\runner~1\appdata\local\temp\pip-install-yixmuu\grpcio\setup.py", line 541, in <module>
        cmdclass=COMMAND_CLASS,
      File "C:\hostedtoolcache\windows\Python\2.7.18\x64\lib\site-packages\setuptools\__init__.py", line 145, in setup
        return distutils.core.setup(**attrs)
      File "C:\hostedtoolcache\windows\Python\2.7.18\x64\lib\distutils\core.py", line 151, in setup
        dist.run_commands()
      File "C:\hostedtoolcache\windows\Python\2.7.18\x64\lib\distutils\dist.py", line 953, in run_commands
        self.run_command(cmd)
      File "C:\hostedtoolcache\windows\Python\2.7.18\x64\lib\distutils\dist.py", line 972, in run_command
        cmd_obj.run()
      File "C:\hostedtoolcache\windows\Python\2.7.18\x64\lib\site-packages\setuptools\command\install.py", line 61, in run
        return orig.install.run(self)
      File "C:\hostedtoolcache\windows\Python\2.7.18\x64\lib\distutils\command\install.py", line 563, in run
        self.run_command('build')
      File "C:\hostedtoolcache\windows\Python\2.7.18\x64\lib\distutils\cmd.py", line 326, in run_command
        self.distribution.run_command(command)
      File "C:\hostedtoolcache\windows\Python\2.7.18\x64\lib\distutils\dist.py", line 972, in run_command
        cmd_obj.run()
      File "C:\hostedtoolcache\windows\Python\2.7.18\x64\lib\distutils\command\build.py", line 127, in run
        self.run_command(cmd_name)
      File "C:\hostedtoolcache\windows\Python\2.7.18\x64\lib\distutils\cmd.py", line 326, in run_command
        self.distribution.run_command(command)
      File "C:\hostedtoolcache\windows\Python\2.7.18\x64\lib\distutils\dist.py", line 972, in run_command
        cmd_obj.run()
      File "C:\hostedtoolcache\windows\Python\2.7.18\x64\lib\site-packages\setuptools\command\build_ext.py", line 84, in run
        _build_ext.run(self)
      File "C:\hostedtoolcache\windows\Python\2.7.18\x64\lib\distutils\command\build_ext.py", line 340, in run
        self.build_extensions()
      File "c:\users\runner~1\appdata\local\temp\pip-install-yixmuu\grpcio\src\python\grpcio\commands.py", line 288, in build_extensions
        support.diagnose_build_ext_error(self, error, formatted_exception)
      File "c:\users\runner~1\appdata\local\temp\pip-install-yixmuu\grpcio\src\python\grpcio\support.py", line 115, in diagnose_build_ext_error
        "backtrace).\n\n{}".format(formatted))
    commands.CommandError:
    
    We could not diagnose your build failure. If you are unable to proceed, please file an issue at http://www.github.com/grpc/grpc with `[Python install]` in the title; please attach the whole log (including everything that may have appeared above the Python backtrace).
    
    Traceback (most recent call last):
      File "c:\users\runner~1\appdata\local\temp\pip-install-yixmuu\grpcio\src\python\grpcio\commands.py", line 285, in build_extensions
        build_ext.build_ext.build_extensions(self)
      File "C:\hostedtoolcache\windows\Python\2.7.18\x64\lib\distutils\command\build_ext.py", line 449, in build_extensions
        self.build_extension(ext)
      File "C:\hostedtoolcache\windows\Python\2.7.18\x64\lib\site-packages\setuptools\command\build_ext.py", line 205, in build_extension
        _build_ext.build_extension(self, ext)
      File "C:\hostedtoolcache\windows\Python\2.7.18\x64\lib\distutils\command\build_ext.py", line 499, in build_extension
        depends=ext.depends)
      File "C:\hostedtoolcache\windows\Python\2.7.18\x64\lib\distutils\msvc9compiler.py", line 473, in compile
        self.initialize()
      File "C:\hostedtoolcache\windows\Python\2.7.18\x64\lib\distutils\msvc9compiler.py", line 383, in initialize
        vc_env = query_vcvarsall(VERSION, plat_spec)
      File "C:\hostedtoolcache\windows\Python\2.7.18\x64\lib\site-packages\setuptools\msvc.py", line 147, in msvc9_query_vcvarsall
        return EnvironmentInfo(arch, ver).return_env()
      File "C:\hostedtoolcache\windows\Python\2.7.18\x64\lib\site-packages\setuptools\msvc.py", line 1227, in return_env
        [self.VCIncludes,
      File "C:\hostedtoolcache\windows\Python\2.7.18\x64\lib\site-packages\setuptools\msvc.py", line 876, in VCIncludes
        return [os.path.join(self.si.VCInstallDir, 'Include'),
      File "C:\hostedtoolcache\windows\Python\2.7.18\x64\lib\site-packages\setuptools\msvc.py", line 555, in VCInstallDir
        raise distutils.errors.DistutilsPlatformError(msg)
    DistutilsPlatformError: Microsoft Visual C++ 9.0 is required. Get it from http://aka.ms/vcpython27
    
    ----------------------------------------
ERROR: Command errored out with exit status 1: 'C:\hostedtoolcache\windows\Python\2.7.18\x64\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'c:\\users\\runner~1\\appdata\\local\\temp\\pip-install-yixmuu\\grpcio\\setup.py'"'"'; __file__='"'"'c:\\users\\runner~1\\appdata\\local\\temp\\pip-install-yixmuu\\grpcio\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'c:\users\runner~1\appdata\local\temp\pip-record-ezdbm8\install-record.txt' --single-version-externally-managed --compile --install-headers 'C:\hostedtoolcache\windows\Python\2.7.18\x64\Include\grpcio' Check the logs for full command output.
Error: Process completed with exit code 1.
```

--- a/build/test-requirements.txt
+++ b/build/test-requirements.txt
@@ -18,7 +18,7 @@ django
 isort
 
 # Integrated TensorBoard tests
-tensorboard
+tensorboard ; python_version > '2.7'
 
 # Python 2.7 support.
 pytest==4.6.9 ; python_version == '2.7'

--- a/news/2 Fixes/17325.md
+++ b/news/2 Fixes/17325.md
@@ -1,0 +1,1 @@
+For CI ensure `tensorboard` is installed in python 3 environments only.

